crates/sui-config/src/builder.rs

```diff
@@ -219,6 +219,7 @@ impl<R: rand::RngCore + rand::CryptoRng> ConfigBuilder<R> {
                         network_address,
                         narwhal_primary_address: validator.narwhal_primary_address.clone(),
                         narwhal_worker_address: validator.narwhal_worker_address.clone(),
+                        narwhal_own_worker_address: validator.narwhal_own_worker_address.clone(),
```

Comment:
Maybe we do need the internal keyword (`narwhal worker_internal_address`) as @huitseeker suggested because the field above is also this validators own worker address so it will be a little confusing.

Reply:
Changed by popular demand :)